src/hs_checkpoint_reader.c

```diff
@@ -257,3 +259,70 @@ void hs_output_checkpoints(hs_checkpoint_reader *cpr, FILE *fh)
   }
   pthread_mutex_unlock(&cpr->lock);
 }
+
+
+void hs_remove_checkpoint(hs_checkpoint_reader *cpr,
+                                const char *key)
+{
+  pthread_mutex_lock(&cpr->lock);
+  lua_pushstring(cpr->values, key);
+  lua_pushnil(cpr->values);
+  lua_settable(cpr->values, LUA_GLOBALSINDEX);
+  pthread_mutex_unlock(&cpr->lock);
+  hs_log(NULL, g_module, 6, "checkpoint removed: %s", key);
+}
+
+
+void hs_cleanup_checkpoints(hs_checkpoint_reader *cpr,
+                            const char *run_path,
+                            int analysis_threads)
+{
+  const char *key;
+  int analysis_thread;
+  char name[HS_MAX_PATH];
+  char path[HS_MAX_PATH];
+  int idx = 0;
+  int max_keys = 64;  // eventually, this should be dynamic
+  char old_keys[max_keys][HS_MAX_PATH];
```

Comment:
Ok, I thought that the mutex lock would not allow to remove the entry during iteration, but actually this is the same thread, so it should work.